com.unity.ml-agents.extensions/Documentation~/Match3.md

```diff
@@ -0,0 +1,60 @@
+# Match-3 Game Support
+
+We provide some utilities to make integration with Match-3 games.
+
+## AbstractBoard class
+The AbstractBoard is the bridge between ML-Agents and your game. It allows ML-Agents to
+* ask your game what the "color" at a row and column is
+* ask your game whether a move is allowed
+* request that your game make a move
+These are handled by implementing the `GetCellType()`, `IsMoveValid()`, and `MakeMove()` abstract methods.
+
+The AbstractBoard also tracks the number of rows, columns, and potential piece types that the board can have.
+
+### `public abstract int GetCellType(int row, int col)`
+Returns the "type" of piece at the given row and column.
+This should be between 0 and NumCellTypes-1 (inclusive).
+
+### `public abstract bool IsMoveValid(Move m)`
+Check whether the particular Move is valid for the game.
+The actual results will depend on the rules of the game, but we provide SimpleIsMoveValid()
+that handles basic match3 rules with no special or immovable pieces.
+
+### `public abstract bool MakeMove(Move m)`
+Instruct the game to make the given move. Returns true if the move was made.
+Note that during training, a move that was marked as invalid may occasionally still be
+requested. If this happens, it is safe to do nothing and request another move.
+
+## Move struct
+The Move struct encapsulates a swap of two adjacent cells. To enumerate over all potential moves:
```

Comment:
```suggestion
The `Move` struct encapsulates a swap of two adjacent cells. To enumerate over all potential moves:
```
It is not obvious to me why this code enumerates all of the Move possible. Maybe add some comments in the code.

Reply:
I'll rewrite this section to be clearer.

Reply:
I also renamed some methods for public consumption:
Move.NumEdgeIndices -> Move.NumPotentialMoves
Move.FromEdgeIndex -> Move.FromMoveIndex